Userland/Utilities/asctl.cpp

```diff
@@ -0,0 +1,203 @@
+            } else if (variable == "r" || variable == "samplerate") {
+                auto sample_rate = arguments[++i].to_int();
+                if (!sample_rate.has_value()) {
+                    warnln("Error: {} is not an integer sample rate", arguments[i - 1]);
+                    return 1;
+                }
+                values_to_set.set(AudioVariable::SampleRate, sample_rate.value());
+            } else {
+                warnln("Error: Unrecognized variable {}", arguments[i]);
+                return 1;
+            }
+        }
+
+        for (auto to_set : values_to_set) {
+            switch (to_set.key) {
+            case AudioVariable::Volume: {
+                int& volume = to_set.value.get<int>();
+                audio_client->set_main_mix_volume(volume);
+                break;
+            }
+            case AudioVariable::Mute: {
+                bool& mute = to_set.value.get<bool>();
+                audio_client->set_muted(mute);
+                break;
+            }
+            case AudioVariable::SampleRate: {
+                int& sample_rate = to_set.value.get<int>();
+                audio_client->set_sample_rate(sample_rate);
+                break;
+            }
+            }
+        }
+    }
+
+    return 0;
+
+    // if (!(mute ^ unmute ^ volume.is_whitespace())) {
+    //     if (human_mode)
+    //         warnln("Only one of mute, unmute or volume must be used");
+    //     return 1;
+    // }
+
+    // if (!mute && !unmute && !volume.is_whitespace()) {
+    //     auto volume = audio_client->get_main_mix_volume();
+    //     if (human_mode)
+    //         outln("Current volume: {:3d}%", volume);
+    //     else
+    //         outln("{}", volume);
+    // }
+    // if (mute) {
+    //     audio_client->set_muted(true);
+    //     if (human_mode)
+    //         outln("Muted.");
+    // } else if (unmute) {
+    //     audio_client->set_muted(false);
+    //     if (human_mode)
+    //         outln("Unmuted.");
+    // } else if (!volume.is_whitespace()) {
+    //     auto new_volume = volume.to_int();
+    //     if (new_volume.has_value()) {
+    //         audio_client->set_main_mix_volume(new_volume.value());
+    //         if (human_mode)
+    //             outln("Set volume to {:3d}%", new_volume.value());
+    //     } else {
+    //         warnln("Error: The volume is not an integer");
+    //         return 1;
+    //     }
+    // }
+
+    // if (!should_print_rate && sample_rate != 0) {
+    //     // Set the sample rate to the specified value
+    //     audio_client->set_sample_rate(sample_rate);
+    //     if (human_mode)
+    //         outln("Set sample rate to {}", sample_rate);
+    // } else {
+    //     // Get and print the sample rate
+    //     sample_rate = audio_client->get_sample_rate();
+    //     if (human_mode)
+    //         outln("Current /dev/audio sample rate: {} Hz", sample_rate);
+    //     else
+    //         outln("{}", sample_rate);
+    // }
```

Comment:
Please don't leave commented-out code in commits.

Reply:
That was dumb, I left the code as a reference and forgot to remove it. -_-